frappe/database/query.py

```diff
@@ -510,8 +512,25 @@ def get_fieldnames_from_child_table(self, doctype, fields):
 
 		return fields
 
-	def set_fields(self, table, fields, **kwargs) -> list:
+	def sanitize_fields(self, fields: str | list | tuple):
+		if isinstance(fields, (list, tuple)):
+			fields = list(fields)
+			for idx, field in enumerate(fields):
+				if isinstance(field, str):
+					field = MARIADB_SPECIFIC_COMMENT.sub(
+						"", sqlparse.format(field, strip_comments=True, keyword_case="lower")
+					)
+					fields[idx] = field
```

Comment:
Why mutating input here? Bad practice in general. 


`#` comments are mariadb specific only. On postgres it's set intersection operator.

Reply:
we're mutating inputs everywhere i think 🙈

https://github.com/frappe/frappe/blob/13e6aca115360a27bb9c198c20606b7b8394f1dc/frappe/database/query.py#L495-L509

https://github.com/frappe/frappe/blob/13e6aca115360a27bb9c198c20606b7b8394f1dc/frappe/database/query.py#L602-L610

Reply:
If it's not required then why do this? 

Sometimes arguments to DB calls are reused later. Just see ERPNext code.

Reply:
It was required, pushed it just to make things work, changed now